src/Stacktrace.php

```diff
@@ -56,42 +56,47 @@ class Stacktrace implements \JsonSerializable
     /**
      * Constructor.
      *
-     * @param ClientInterface $client The Raven client
+     * @param Options $options The client options
      */
-    public function __construct(ClientInterface $client)
+    public function __construct(Options $options)
     {
-        $this->client = $client;
-        $this->serializer = $client->getSerializer();
-        $this->reprSerializer = $client->getRepresentationSerializer();
+        $this->options = $options;
+        $this->serializer = new Serializer($this->options->getMbDetectOrder());
```

Comment:
I started to refactor this now, `Serializer` now only lives on the client.

Reply:
Well done, we should add an interface for the serializer (in another PR, obviously) so that we can work on a new implementation of the serializer (the current one is buggy and too limited in features) that can be switched later at any time if needed